mne/viz/utils.py

```diff
@@ -1032,3 +1036,102 @@ def _plot_sensors(pos, colors, ch_names, title, show_names, show):
     fig.suptitle(title)
     plt_show(show)
     return fig
+
+
+def compute_scalings(scalings, inst):
+    """Compute scalings for each channel type automatically.
+
+    Parameters
+    ----------
+    scalings : dict
+        The scalings for each channel type. If any values are
+        'auto', this will automatically compute a reasonable
+        scaling for that channel type.
+    inst : instance of Raw or Epochs
+        The data you wish to compute scalings for. If Epochs,
+        data will be preloaded. If Raw, a window of max 20
+        seconds will be loaded from the middle of the data for
+        estimating scaling.
+
+    Returns
+    -------
+    scalings : dict
+        A scalings dictionary with updated scalings
+    """
+    from copy import deepcopy
+    from mne.io.base import _BaseRaw
+    from mne.epochs import _BaseEpochs
+    if not isinstance(inst, (_BaseRaw, _BaseEpochs)):
+        raise ValueError('Must supply either Raw or Epochs')
+
+    if scalings == 'auto':
+        # If we want to auto-compute everything
+        scalings = {i_type: 'auto' for i_type in _channel_types + _meg_types}
```

Comment:
This is standard Py3 dict comp.

I'm fairly sure it will also work for 2.6.

But I've been repeatedly told we shouldn't do it like this here, although I don't know why either.